packages/asset-swapper/src/quote_consumers/exchange_proxy_swap_quote_consumer.ts

```diff
@@ -380,91 +421,147 @@ export class ExchangeProxySwapQuoteConsumer implements SwapQuoteConsumerBase {
     ): Promise<string> {
         throw new Error('Execution not supported for Exchange Proxy quotes');
     }
-}
 
-function isDirectSwapCompatible(
-    quote: SwapQuote,
-    opts: ExchangeProxyContractOpts,
-    directSources: ERC20BridgeSource[],
-): boolean {
-    // Must not be a mtx.
-    if (opts.isMetaTransaction) {
-        return false;
-    }
-    // Must not have an affiliate fee.
-    if (!opts.affiliateFee.buyTokenFeeAmount.eq(0) || !opts.affiliateFee.sellTokenFeeAmount.eq(0)) {
-        return false;
-    }
-    // Must not have a positive slippage fee.
-    if (opts.affiliateFee.feeType === AffiliateFeeType.PositiveSlippageFee) {
-        return false;
-    }
-    // Must be a single order.
-    if (quote.orders.length !== 1) {
-        return false;
-    }
-    const order = quote.orders[0];
-    if (!directSources.includes(order.source)) {
-        return false;
-    }
-    // VIP does not support selling the entire balance
-    if (opts.shouldSellEntireBalance) {
-        return false;
+    private _encodeMultiplexBatchFillCalldata(quote: SwapQuote): string {
+        const multiplex = new MultiplexFeatureContract(NULL_ADDRESS, this.provider);
```

Comment:
Not sure how heavy these contract objects are but might ease the load to create them once in the constructor.